opentelemetry/proto/collector/trace/v1/trace_service.proto

```diff
@@ -37,23 +38,33 @@ service TraceService {
 }
 
 message ExportTraceServiceRequest {
-  // Span data. An array of ResourceSpans.
+  // An array of ResourceSpans.
   // For data coming from a single resource this array will typically contain one
   // element. Intermediary nodes (such as OpenTelemetry Collector) that receive
   // data from multiple origins typically batch the data before forwarding further and
   // in that case this array will contain multiple elements.
   repeated ResourceSpans resource_spans = 1;
 }
 
-// A collection of spans from a Resource.
+// A collection of LibrarySpans from a Resource.
 message ResourceSpans {
-  // A list of Spans that originate from a resource.
-  repeated opentelemetry.proto.trace.v1.Span spans = 1;
+  // A list of LibrarySpans that originate from a resource.
+  repeated LibrarySpans library_spans = 1;
```

Comment:
This pattern helps with the specific problem - single process often host many libraries. This pattern can be generalized further and insted of single outing the Library info - we can have enclosed resource objects at multiple levels. So other patterns like springboot multi-app hosting in a single process or collector appending host information to different applications can also be optimized in a similar way. What do you think?

Reply:
That's something that is that should be discussed in the spec or even an OTEP. It sounds useful but also potentially complicated to implement.

Reply:
https://github.com/open-telemetry/oteps/pull/84/files

Reply:
Will this proposal change the implementation? Should it be renamed to component spans?

Reply:
The "component" changed, so we stay with library.